renku/core/models/datasets.py

```diff
@@ -128,6 +128,12 @@ def _convert_dataset_files_creators(value):
         return [Creator.from_jsonld(c) for c in coll]
 
 
+def convert_filename_path(p):
+    """Return name of the file."""
+    if p and len(p) >= 1:
+        return Path(p).name
+
+
```

Comment:
Can you please explain how this fix works? If `filename` is `None` then no `DatasetFile` object will be created!?

Reply:
I think it should be created, but `filename` field will remain None within the instantiated object.

The problem of why raise was happening is because this `Path(None)`  results in `TypeError`.